mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -179,15 +194,21 @@ public EntityDefinition getDefinition(TypeElement classElement) {
   }
 
   private VariableElement findField(
-      TypeElement classElement, String propertyName, TypeMirror fieldType) {
-    for (Element child : classElement.getEnclosedElements()) {
-      if (child.getKind() != ElementKind.FIELD) {
+      Set<TypeElement> typeHierarchy, String propertyName, TypeMirror fieldType) {
+    for (TypeElement classElement : typeHierarchy) {
+      // skip interfaces as they can't have fields
+      if (classElement.getKind().isInterface()) {
         continue;
       }
-      VariableElement field = (VariableElement) child;
-      if (field.getSimpleName().toString().equals(propertyName)
-          && context.getTypeUtils().isSameType(field.asType(), fieldType)) {
-        return field;
+      for (Element child : classElement.getEnclosedElements()) {
+        if (child.getKind() != ElementKind.FIELD) {
+          continue;
+        }
+        VariableElement field = (VariableElement) child;
+        if (field.getSimpleName().toString().equals(propertyName)
+            && context.getTypeUtils().isSameType(field.asType(), fieldType)) {
+          return field;
+        }
       }
     }
     return null;
```

Comment:
this would mean passing optionals between methods, which is kinda ugly, but I see your point.  I agree that it's not good to be consistent though.  I'll change to standardize on not using Optional and using @Nullable where applicable.

Reply:
actually looking at this more, the two places we use Optional, we pass in as constructors to `DefaultEntityDefinition`.  I'm going to leave that be since it isn't relevant for this PR.   I will annotate arguments as nullable though where applicable.